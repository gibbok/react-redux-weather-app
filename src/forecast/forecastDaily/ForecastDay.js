// @flow
import React from 'react'
import moment from 'moment'
import IconWeather from '../../shared/icon/IconWeather'

const ForecastDay = ({ date, tempMin, tempMax, iconCode, weatherDescription }:{+date:Date, +tempMin: number, +tempMax: number, +iconCode:string, +weatherDescription:string }) => {
  const dateFormat = moment.unix(date).format('ddd, MMM D')
  const tempMinRounded = Math.round(tempMin)
  const tempMaxRounded = Math.round(tempMax)
  return (
    <div>
      <div>{dateFormat}</div>
      <div>
        <IconWeather code={iconCode} />
      </div>
      <div>
        <div>
          {tempMinRounded}&#176;
        </div>
        <div>
          {tempMaxRounded}&#176;
        </div>
      </div>
      <div>
        {weatherDescription}
      </div>
    </div>
  )
}
export default ForecastDay
