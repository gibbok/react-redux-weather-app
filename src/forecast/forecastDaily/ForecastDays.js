
// @flow
import * as React from 'react'
import ForecastDay from './ForecastDay'

/* eslint-disable no-undef */
type PropsType = {
  +forecastDays: Array<Object>,
  +onForecastDayClick: Object
}
/* eslint-enable no-undef */

const ForecastDays:React.StatelessFunctionalComponent<PropsType> = ({ forecastDays, onForecastDayClick }:PropsType):React.Element<any> => (
  <div>
    <h2>Daily</h2>
    <div>
      {forecastDays.map((day, index) =>
        <ForecastDay
          key={index}
          {...forecastDays[index]}
      />
    )}
    </div>
  </div>
)

export default ForecastDays
