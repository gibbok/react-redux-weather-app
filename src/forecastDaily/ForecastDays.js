import React from 'react'
import ForecastDay from './ForecastDay'

const ForecastDays = ({ forecastDays, onForecastDayClick }) => (
  <div>
    <h2>Daily</h2>
    <ul>
      {forecastDays.map((day, index) =>
        <ForecastDay
          key={index}
          {...forecastDays[index]}
          onClick={() => onForecastDayClick(day)}
      />
    )}
    </ul>
  </div>
)

export default ForecastDays
