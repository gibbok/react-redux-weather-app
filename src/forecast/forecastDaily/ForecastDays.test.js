// @flow
import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import ForecastDays from './ForecastDays'
import ForecastDay from './ForecastDay'

describe('<ForecastDays />', () => {
  it('should render', () => {
    const forecastDays = [
      {
        date: 1505145600,
        dt: 1505145600,
        iconCode: 801,
        momentId: '2017:8:11:1',
        tempMax: 15.57,
        tempMin: 11.20,
        weatherDescription: 'few clouds'
      },
      {
        date: 1505165700,
        dt: 1505165700,
        iconCode: 200,
        momentId: '2017:8:11:2',
        tempMax: 14.12,
        tempMin: 12.40,
        weatherDescription: 'storm rains'
      }
    ]
    const wrapper = shallow(
      <ForecastDays
        forecastDays={forecastDays}
        onForecastDayClick={() => {}}
        />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should contain two ForecastDays components', () => {
    const forecastDays = [
      {
        date: 1505145600,
        dt: 1505145600,
        iconCode: 801,
        momentId: '2017:8:11:1',
        tempMax: 15.57,
        tempMin: 11.20,
        weatherDescription: 'few clouds'
      },
      {
        date: 1505165700,
        dt: 1505165700,
        iconCode: 200,
        momentId: '2017:8:11:2',
        tempMax: 14.12,
        tempMin: 12.40,
        weatherDescription: 'storm rains'
      }
    ]
    const wrapper = shallow(
      <ForecastDays
        forecastDays={forecastDays}
        onForecastDayClick={() => {}}
          />
      )
    expect(wrapper.find(ForecastDay).length).toBe(2)
  })
})
