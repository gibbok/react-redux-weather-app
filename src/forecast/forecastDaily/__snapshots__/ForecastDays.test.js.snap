// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastDays /> should render 1`] = `
<div>
  <h2>
    Daily
  </h2>
  <div>
    <ForecastDay
      date={1505145600}
      dt={1505145600}
      iconCode={801}
      momentId="2017:8:11:1"
      tempMax={15.57}
      tempMin={11.2}
      weatherDescription="few clouds"
    />
    <ForecastDay
      date={1505165700}
      dt={1505165700}
      iconCode={200}
      momentId="2017:8:11:2"
      tempMax={14.12}
      tempMin={12.4}
      weatherDescription="storm rains"
    />
  </div>
</div>
`;
